mailbox/tools/indexer/src/main/java/org/apache/mailbox/tools/indexer/ReprocessingContextInformationDTO.java

```diff
@@ -0,0 +1,137 @@
+            this.mailboxId = mailboxId;
+            this.uids = uids;
+        }
+
+        public String getMailboxId() {
+            return mailboxId;
+        }
+
+        public List<Long> getUids() {
+            return uids;
+        }
+    }
+
+    public static final Function<TaskType, Function<MailboxId.Factory, AdditionalInformationDTOModule<ReprocessingContextInformation, ReprocessingContextInformationDTO>>> SERIALIZATION_MODULE =
+        taskType -> factory ->
+            DTOModule.forDomainObject(ReprocessingContextInformation.class)
+                .convertToDTO(ReprocessingContextInformationDTO.class)
+                .toDomainObjectConverter(dto -> new ReprocessingContextInformation(dto.successfullyReprocessedMailCount, dto.failedReprocessedMailCount, deserializeFailures(factory, dto.failures)))
+                .toDTOConverter((details, type) -> new ReprocessingContextInformationDTO(details.getSuccessfullyReprocessedMailCount(), details.getFailedReprocessedMailCount(), serializeFailures(details.failures())))
+                .typeName(taskType.asString())
+                .withFactory(AdditionalInformationDTOModule::new);
+
+    static ReIndexingExecutionFailures deserializeFailures(MailboxId.Factory mailboxIdFactory,
+                                                                   List<ReindexingFailureDTO> failures) {
+        List<ReIndexingExecutionFailures.ReIndexingFailure> reIndexingFailures = failures
+            .stream()
+            .flatMap(failuresForMailbox ->
+                getReIndexingFailureStream(mailboxIdFactory, failuresForMailbox))
+            .collect(Guavate.toImmutableList());
+
+        return new ReIndexingExecutionFailures(reIndexingFailures);
+    }
+
+    private static Stream<ReIndexingExecutionFailures.ReIndexingFailure> getReIndexingFailureStream(MailboxId.Factory mailboxIdFactory, ReindexingFailureDTO failuresForMailbox) {
+        return failuresForMailbox.uids
+            .stream()
+            .map(uid ->
+                new ReIndexingExecutionFailures.ReIndexingFailure(mailboxIdFactory.fromString(failuresForMailbox.mailboxId), MessageUid.of(uid)));
+    }
+
+
+    static List<ReindexingFailureDTO> serializeFailures(ReIndexingExecutionFailures failures) {
+        ImmutableListMultimap<MailboxId, ReIndexingExecutionFailures.ReIndexingFailure> failuresByMailbox = failures.failures()
+            .stream()
+            .collect(Guavate.toImmutableListMultimap(ReIndexingExecutionFailures.ReIndexingFailure::getMailboxId));
+
+        return failuresByMailbox
+            .asMap()
+            .entrySet()
+            .stream()
+            .map(failureByMailbox ->
+                new ReindexingFailureDTO(
+                    failureByMailbox.getKey().serialize(),
+                    extractMessageUidsFromFailure(failureByMailbox))
+            ).collect(Guavate.toImmutableList());
```

Comment:
ident?

Reply:
> Is it me or we missed task serialization for all reIndexing tasks?
> 
> What do we plan to avoid such mistakes in the future?

The serialization of the task was done. We just extracted the Task DTO in this PR.